alvr/graphics/src/stream.rs

```diff
@@ -470,3 +495,9 @@ pub fn foveated_encoding_shader_constants(
 
     (optimized_view_resolution_aligned.as_uvec2(), constants)
 }
+
+pub fn scale_resolution(resolution: UVec2, upscaling: UpscalingConfig) -> UVec2 {
```

Comment:
You should take upscaling by reference so you don't need to clone it. Also my intent with this function was another, call it "compute_target_view_resolution", which now only takes the upscaling parameters but in the future it could take more parameters